libavformat/hls.c

```diff
@@ -2287,6 +2291,26 @@ static int hls_read_packet(AVFormatContext *s, AVPacket *pkt)
 
         *pkt = pls->pkt;
         pkt->stream_index = st->index;
+
+        /* Playlist metadata */
+        av_dict_set_int(&metadata_dict, "targetDuration", pls->target_duration, 0);
+        av_dict_set_int(&metadata_dict, "playlistType", pls->type, 0);
+        av_dict_set_int(&metadata_dict, "variants", c->n_variants, 0);
+        av_dict_set_int(&metadata_dict, "bandwidth", c->variants[pls->index]->bandwidth, 0);
```

Comment:
Although this is probably safe, I wonder if I should check that pls->index < n_variants before doing this?

Reply:
Might as well